js/ai/src/generate/action.ts

```diff
@@ -52,15 +50,10 @@ import {
   Part,
   Role,
   ToolDefinitionSchema,
-  ToolResponsePart,
   resolveModel,
 } from '../model.js';
-import {
-  ToolAction,
-  ToolInterruptError,
-  resolveTools,
-  toToolDefinition,
-} from '../tool.js';
+import { ToolAction, resolveTools, toToolDefinition } from '../tool.js';
+import { assertValidToolNames, resolveToolRequests } from './tool-loop.js';
 
 export const GenerateUtilParamSchema = z.object({
```

Comment:
`GenerateUtilParam` is not a very good name, is it? This is a great opportunity to rename this... before it becomes part of the cross-runtime contract (when we expose it to the dev ui).

How about GenerateActionOptions? A nod to GenerateOptions...

Reply:
Yup, meant to do that actually and forgot. Done!